pkg/webhook/mutatingwebhook.go

```diff
@@ -0,0 +1,136 @@
+package webhook
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"golang.org/x/net/context"
+
+	admregv1 "k8s.io/api/admissionregistration/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// MutatingWebhookConfigurationBuilder provides struct for MutatingWebhookConfiguration object which contains connection
+// to cluster and MutatingWebhookConfiguration definition.
+type MutatingWebhookConfigurationBuilder struct {
+	// MutatingWebhookConfiguration definition. Used to create MutatingWebhookConfiguration object.
+	Definition *admregv1.MutatingWebhookConfiguration
+	// Created MutatingWebhookConfiguration object.
+	Object *admregv1.MutatingWebhookConfiguration
+	// Used in functions that define or mutate MutatingWebhookConfiguration definitions.
+	// errorMsg is processed before MutatingWebhookConfiguration object is created.
+	errorMsg string
+	// apiClient opens api connection to the cluster.
+	apiClient *clients.Settings
+}
+
+// PullMutatingWebhookConfiguration pulls existing MutatingWebhookConfiguration from cluster.
+func PullMutatingWebhookConfiguration(apiClient *clients.Settings, name string) (
+	*MutatingWebhookConfigurationBuilder, error) {
+	glog.V(100).Infof("Pulling existing MutatingWebhookConfiguration name %s from cluster", name)
+
+	builder := MutatingWebhookConfigurationBuilder{
+		apiClient: apiClient,
+		Definition: &admregv1.MutatingWebhookConfiguration{
+			ObjectMeta: metaV1.ObjectMeta{
+				Name: name,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the MutatingWebhookConfiguration is empty")
+
+		builder.errorMsg = "MutatingWebhookConfiguration 'name' cannot be empty"
+	}
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("MutatingWebhook object %s doesn't exist", name)
+	}
+
+	builder.Definition = builder.Object
+
+	return &builder, nil
+}
+
+// Exists checks whether the given MutatingWebhookConfiguration object exists in the cluster.
+func (builder *MutatingWebhookConfigurationBuilder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	var err error
+	builder.Object, err = builder.Get()
+
+	return err == nil || !k8serrors.IsNotFound(err)
+}
+
+// Get returns MutatingWebhookConfiguration object.
+func (builder *MutatingWebhookConfigurationBuilder) Get() (*admregv1.MutatingWebhookConfiguration, error) {
+	if valid, err := builder.validate(); !valid {
+		return &admregv1.MutatingWebhookConfiguration{}, err
+	}
+
+	mutatingWebhookConfiguration := &admregv1.MutatingWebhookConfiguration{}
+	err := builder.apiClient.Get(context.TODO(), goclient.ObjectKey{
+		Name: builder.Definition.Name,
+	}, mutatingWebhookConfiguration)
+
+	if err != nil {
```

Comment:
I'd log the error to make it easier to troubleshoot

Reply:
added